src/coord/src/catalog.rs

```diff
@@ -293,17 +298,12 @@ impl CatalogItem {
     pub fn uses(&self) -> Vec<GlobalId> {
         match self {
             CatalogItem::Func(_) => vec![],
-            CatalogItem::Index(idx) => vec![idx.on],
-            CatalogItem::Sink(sink) => vec![sink.from],
+            CatalogItem::Index(idx) => idx.depends_on.clone(),
+            CatalogItem::Sink(sink) => sink.depends_on.clone(),
             CatalogItem::Source(_) => vec![],
-            CatalogItem::Table(_) => vec![],
-            CatalogItem::Type(typ) => match &typ.inner {
-                TypeInner::Array { element_id } => vec![*element_id],
-                TypeInner::Base | TypeInner::Pseudo => vec![],
-                TypeInner::List { element_id } => vec![*element_id],
-                TypeInner::Map { key_id, value_id } => vec![*key_id, *value_id],
-            },
-            CatalogItem::View(view) => view.optimized_expr.as_ref().global_uses(),
+            CatalogItem::Table(table) => table.depends_on.clone(),
+            CatalogItem::Type(typ) => typ.depends_on.clone(),
+            CatalogItem::View(view) => view.depends_on.clone(),
```

Comment:
I think this can just return `&[GlobalId]` now, and then you can avoid the clone!